File: 01/python/main.py
from python_utils.input import add_argparse, advent_input

get_int = lambda x: int(x.strip())


@advent_input('01/input_1.txt')
def task_one(lines):
    prev_value = get_int(next(lines))
    count = 0

    for line in lines:
        if (current_value := get_int(line)) > prev_value:
            count += 1

        prev_value = current_value

    print(count)


@advent_input('01/input_1.txt')
def task_two(lines):
    values = [get_int(line) for line in lines]

    prev = 0
    count = 0
    for i in range(values_length := len(values)):
        if (i + 3) >= values_length:
            break 
        
        if (current_value := sum(values[i:i+3])) > prev:
            count += 1
        
        prev = current_value

    print(count)

@add_argparse
def main(task):
    task_one() if task == "one" else task_two()


if __name__ == "__main__":
    main()

File: 01/cpp/main.cpp
#include <filesystem>
#include <fstream>
#include <string>
#include <iostream>
#include <vector>

#include <utils.h>


void one(){
    std::vector<int> values = textUtils::getFileLinesInt("02/input_1.txt");

    int prev = values.at(0);
    int count = 0;

    for (int i: values) {
        if (i > prev) {
            count++;
        }

        prev = i;
    }

    std::cout << count << "\n";
}

void two(){
    std::vector<int> values = textUtils::getFileLinesInt("01/input_1.txt");

    int prev = 0;
    int count = 0;

    for (int i = 0; i < values.size(); i++) {
        if (i + 3 >= values.size()) {
            break;
        }

        int a = values.at(i);
        int b = values.at(i + 1);
        int c = values.at(i + 2);

        int sum = a + b + c;

        if (sum > prev) {
           count++;
        }

        prev = sum;
    }

    std::cout << count << "\n";
}

int main() {
    one();
    two();
}